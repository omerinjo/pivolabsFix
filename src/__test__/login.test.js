import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { BrowserRouter as Router } from 'react-router-dom';

import Login from '../screens/login/login'

const initialState = {};

const mockStore = configureStore();
let store;

beforeEach(() => {
    //creates the store with any initial state or middleware needed  
    store = mockStore(initialState)
})

describe('<Login> rendering', () => {
    test('render correctly', () => {
        const component = renderer.create(<Router><Login store={store} /> </Router>)
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
}) 