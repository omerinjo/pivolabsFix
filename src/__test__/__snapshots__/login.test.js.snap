// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> rendering render correctly 1`] = `
Array [
  <div
    className="justify-content-md-center omer container"
  >
    <nav
      className="navbar-container navbar navbar-expand-lg navbar-light bg-light"
      data-test="guestNavigation"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login 
        </a>
        <a
          className="nav-link"
          href="/standing"
          onClick={[Function]}
        >
          Standing 
        </a>
      </div>
    </nav>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formGroupEmail"
        >
          Email address
        </label>
        <input
          className="form-control"
          id="formGroupEmail"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
          htmlFor="formGroupPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="formGroupPassword"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  " ",
]
`;
